### ISSUE TITLE
Compiling issue: Clang with libomp on MacOS

### ISSUE THREAD

**Issue by fishjojo** [pyscf/pyscf#940]
When pyscf is compiled with `Clang` and linked with `libomp` (installed through homebrew), I need to explicitly add `-lomp` to the linker flags, e.g., 
```
set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -lomp")
```
Otherwise, errors of `Undefined symbols` appear:
```
Undefined symbols for architecture x86_64:
  "___kmpc_barrier", referenced from:
      _.omp_outlined. in transpose.c.o
      _.omp_outlined..2 in transpose.c.o
      _.omp_outlined..4 in transpose.c.o
      _.omp_outlined..6 in transpose.c.o
      _.omp_outlined. in pack_tril.c.o
      _.omp_outlined..2 in pack_tril.c.o
      _.omp_outlined..4 in pack_tril.c.o
      ...
```
What can be the possible cause of this?
Using `gcc` with `libgomp` has no issues.
My cmake version is 3.19.

**Comment by fishjojo** [comment]
Please see the following:
https://github.com/fishjojo/pyscf/blob/0410cd950f1d228ebd3597b1f394fa4db4aa630e/pyscf/lib/CMakeLists.txt#L74-L76

I think this problem is due to cmake, which doesn't write the correct compiling flags.

**Comment by sunqm** [comment]
> Please see the following: https://github.com/fishjojo/pyscf/blob/0410cd950f1d228ebd3597b1f394fa4db4aa630e/pyscf/lib/CMakeLists.txt#L74-L76
> 
> I think this problem is due to cmake, which doesn't write the correct link flags.

Do you plan to merge this patch to the main branch?

**Comment by fishjojo** [comment]
> > Please see the following: https://github.com/fishjojo/pyscf/blob/0410cd950f1d228ebd3597b1f394fa4db4aa630e/pyscf/lib/CMakeLists.txt#L74-L76
> > I think this problem is due to cmake, which doesn't write the correct link flags.
> 
> Do you plan to merge this patch to the main branch?

I'm not sure if this is the best way to solve the problem...

**Comment by sunqm** [comment]
Fixed in PR #1249
